Footer

zijunw04 opened the pull request:
Added 
- footer 
- scroll animation when user click shop now
- added a temporary product list to show footers
- changed all the link to open a new tab instead of changing tab from original
- make the footer color and button text color interact with light/dark mode button

Commit:
added Hero section of the website, changed the contributor page to when user clicks contact, added custom color for the theme of the website, added text shadow to the theme, changed the logo of sun/moon so they are white instead of black.

Commit:
Merge branch 'main' of https://github.com/acm-uic/marketplace into zijunw-contribute

Commit:
changed the theme to match Figma

Commit:
fixed the layout of the button

Commit:
footer, fix on color match, scroll animation

Commit:
fixed github directing wrong page

Commit:
updated footer + product list match figma design

Ajknight121 commented on app/components/header.jsx:
```diff
@@ -8,21 +8,21 @@ import moon from "../images/moon.png"
-                <Link className="text-2xl text-white hover:underline w" href='https://acm.uic.edu/about'>Contact</Link>
+                <Link href="/" className="text-2xl ml-6 mr-10 text-white hover:underline">Shop</Link>
+                <Link className="text-2xl mr-10 text-white hover:underline" href="https://acm.uic.edu/about" target="_blank">About</Link>
+                <Link className="text-2xl text-white hover:underline w" href='/Contributors'>Contact</Link>
```
This link should still be to the ACM contact page as it was before. The website contributor link will be in the footer.

Commit:
changed the contact back to acm.uic.edu/contact

Ajknight121 commented on app/components/footer.jsx:
Also add the ACM Linkedin (https://www.linkedin.com/company/acm-cs-uic/) and Instagram (https://www.instagram.com/acmatuic/)

Ajknight121 commented:
Awesome work! The scroll transition looks good and the colors are correct. This pull should also complete the hero pull request.

Commit:
added instagram and linkedin to footer

Ajknight121 commented on app/components/footer.jsx:
Add link called "Website Credit & Contributors" to the footer that navigates to the "/Contributors" page. I've added an example to the Figma

Commit:
Added contributor in the footer

Commit:
added header and footer to the contributor page

zijunw04 commented:
since i did the navigates to the "/Contributors" page, I decided to change the contributor page a bit to fit the Figma design. 

I added
- header to contributor page with inverted color from original
- footer to contributor page 
- a title to contributor page
- removed the background color
- when user click on github @ on contributor, it goes to new tab instead of opening it directly on the tab

but there is something that I could not figure out what to do is the 
- syncing light/dark mode from the shop page to contributor page (light/dark still work, it just resets so it does not cross over)

let me know if there is anything I need to revert or change!

Ajknight121 commented on app/Contributors/page.jsx:
```diff
@@ -1,24 +1,32 @@
+"use client"
```
Remove state and header/footer

Ajknight121 commented on app/page.jsx:
Remove Header and Footer from this page.

zijunw04 replied:
Can I leave the header/footer in the main page for easy navigation and that could be removed once the layout.jsx was implemented? Everything else is resolved on the new commit

I didnt want to mess anything up because there are lots of value tight to the header like isDarkMode and setIsDarkMode and background color is dependent on it too

Ajknight121 commented on app/page.jsx:
```diff
@@ -1,19 +1,29 @@
+import HeroScreen from "./components/hero"
+import Footer from "./components/footer"
+import ProductList from "./components/productlist";
+import { useEffect, useState, useRef } from "react";
```
remove useEffect

Ajknight121 reviewed (changes requested):
The Header and Footer should actually be removed from all pages and added layout.jsx. Components added to layout.jsx should render on all other pages. Moving Header and footer to layout.jsx should solve the issues of keeping the same theme applied. This also could have been solved with React's useContext. I'll go over context in the next meeting.

Commit:
revert the contributor page + remove useEffect

Commit:
fixed header

Ajknight121 reviewed (approved):
Spectacular work!
